transport/src/main/java/io/netty5/channel/socket/nio/NioServerSocketChannel.java

```diff
@@ -223,7 +236,16 @@ protected ServerSocketChannel javaChannel() {
 
     @Override
     protected SocketAddress localAddress0() {
-        return SocketUtils.localSocketAddress(javaChannel().socket());
+        try {
+            SocketAddress address = javaChannel().getLocalAddress();
+            if (isDomainSocket(family)) {
+                address = toDomainSocketAddress(address);
+            }
+            return address;
+        } catch (IOException e) {
+            // Just return null
+            return null;
```

Comment:
Could it be worth debug-logging the exception?

Reply:
The Socket implementation that we used before returns `new InetSocketAddress(InetAddress.getAnyAddress() ...` in this case (and not log). Maybe we should just do this as well. ? @chrisvest WDYT ?

Reply:
Yeah, I think returning `null` is fine.